mesonbuild/optinterpreter.py

```diff
@@ -73,7 +73,10 @@ def __init__(self, subproject, command_line_options):
         self.subproject = subproject
         self.cmd_line_options = {}
         for o in command_line_options:
-            (key, value) = o.split('=', 1)
+            try:
+                (key, value) = o.split('=', 1)
+            except ValueError:
+                raise OptionException('Option {!r} must have a value separated by equals sign.'.format(o))
```

Comment:
Well, it's good to see the length check is correct now; I'm not totally happy with the phrasing, so I can fix that later if there are any better suggestions.